compiler/codegen/codegen.cpp

```diff
@@ -2161,6 +2161,18 @@ static void setupDefaultFilenames() {
       strncat(executableFilename, lastSlash,
               sizeof(executableFilename)-strlen(executableFilename)-1);
 
+      if (fLibraryPython) {
+        strncpy(pythonModulename, lastSlash, sizeof(pythonModulename)-1);
```

Comment:
while lastSlash describes how it was computed, this value *really* needs a different variable name. I'd recommend e.g.
``` c
     const char* lastSlash = strrchr(mainModFilename, '/');
     const char* filename = NULL;
     if (lastSlash == NULL) {
       filename = mainModFilename;
     } else {
       filename = lastSlash + 1;
    }
```

and replacing your uses of lastSlash with filename.